### PR TITLE
feat: add section on structs

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@dghosef I would be happy with some feedback when you have time :)

If nothing else, typos are good to find.

File: src/structs.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+        u->id = 0
+}
+```
+
+The Flix compiler emits a compiler error:
+
+```
+❌ -- Resolution Error -------------------------------------------------- 
+
+>> Modification of immutable field 'id' on User'.
```
Comment (dghosef) [inline src/structs.md:214]: The error message when I run it says:

```
-- Resolution Error -------------------------------------------------- /Users/dghosef/flix/main/test/flix/test.flix

>> Modification of immutable field `User.id`. Mark the field as `mut` to allow mutation.

8 |         u->id = 0
               ^^
               field not marked `mut`
```

**Review by dghosef — COMMENTED** [review]
Looks good to me, just 2 minor things I've noticed

File: src/structs.md (hunk 1 above)
Comment (magnus-madsen) [reply src/structs.md:214]: I updated the error messages.

File: src/structs.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
+## Structs
+
+> **Note:** Requires Flix version 0.51.0
+
+Flix supports mutable _scoped_ structs. In Flix, like with arrays, every struct
+must have an associated region. Flix supports three operations on structs:
+
+- Creating a struct with `new Struct @ rc { ... }`.
+- Accessing the field of a struct with `struct->field`.
+- Updating a _mutable_ field of a struct with `struct-> field = ...`.
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply src/structs.md:10]: Good catch. I will update it when I get around to it.

**Pull request merged**